### PR TITLE
Fix crash

### PR DESCRIPTION
**Pull request by mtrew2015**
# Description

Fixes the issues that were causing the app to crash on load... restructures a couple of queries that were dependent only on Firebase, and connects them to Prisma.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

**Comment by mtrew2015** [comment]
[Authentication Bugs](https://trello.com/c/o4RLwvsN/97-authentication-bugs)

File: client/src/App.js [hunk 1]
```diff
@@ -318,101 +318,6 @@ class App extends Component {
...
-                  );
-                });
-              } else return <h1>No Data</h1>;
-            }}
-          </Query>
-        </div>
-      </Router>
-    );
-  }
 }
```
Comment (TheGameFreak720) [inline client/src/App.js]: I think this is a trailing bracket from all you removed

File: client/src/components/Account/ProjectCard/ProjectCard.js [hunk 2]
```diff
@@ -144,7 +144,13 @@ class ProjectCard extends React.Component {
                   })}
                   <h2>Reviews:</h2>
                   {this.state.reviews.map((rev) => {
-                    return <ReviewCard key={rev.id} review={rev} />;
+                    return (
+                      <ReviewCard
+                        key={rev.id}
```
Comment (TheGameFreak720) [inline client/src/components/Account/ProjectCard/ProjectCard.js]: Are these cards more complicated than the ones we got on Home that require to be a separate component?

File: client/src/components/Account/ReviewCard/ReviewCard.js [hunk 3]
```diff
@@ -8,15 +8,15 @@ class ReviewCard extends React.Component {
...
+    const { users, review } = this.props;
 
     const json = localStorage.getItem('authUser');
     const authUser = JSON.parse(json);
     let visitor = [];
     let loggedIn = false;
     if (authUser !== null)
       visitor = users.filter((u) => u.email === authUser.email)[0];
-    console.log({ users: users });
+    console.log({ reviewCardUsers: users, visitor: visitor });
```
Comment (TheGameFreak720) [inline client/src/components/Account/ReviewCard/ReviewCard.js]: Remove the log please

File: client/src/components/SignIn/ThirdPartyApi/Github.js [hunk 4]
```diff
@@ -61,99 +76,128 @@ class SignInGithubBase extends Component {
...
+                  <button type="submit" disabled>
+                    {' '}
+                    Sign In with Github{' '}
+                  </button>{' '}
+                  <div>Loading...</div>
+                  {error && <p> {error.message} </p>}
+                </form>
+              );
+            if (checkError) {
+              console.log({ error: checkError });
```
Comment (TheGameFreak720) [inline client/src/components/SignIn/ThirdPartyApi/Github.js]: Logs like this one is fine since we don't have a way to show errors outside the console

**Comment by TheGameFreak720** [comment]
Good work guys, I saw a lot of cleanup and that is always nice. I left a few questions and suggestions here and there but everything looks good

File: client/src/components/Account/ProjectCard/ProjectCard.js (hunk 2 above)
Comment (alasalle) [reply client/src/components/Account/ProjectCard/ProjectCard.js]: Review Cards are more complicated, yes.

Comment (TheGameFreak720) [reply client/src/components/Account/ProjectCard/ProjectCard.js]: In what sense? What makes them more complicated?

Comment (alasalle) [reply client/src/components/Account/ProjectCard/ProjectCard.js]: Review cards have their own state and handle the functionality for reviews; allowing same user to edit/allowing others to like and dislike. The modal requirements are also fulfilled in the review card component. Featured cards are purely presentation focused. When a user clicks on the review from the project card, the modal should pop up w/ all the bells and whistles associated w/ reviews.

**Pull request merged**